controllers/storagecluster/reconcile.go

```diff
@@ -433,6 +433,15 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			message := fmt.Sprintf("Error while reconciling: %v", returnErr)
 			statusutil.SetErrorCondition(&instance.Status.Conditions, reason, message)
 
+			// if the error was due to skipped storage classes, we shouldn't set the phase to error
+			// instead of error phase we will set it to progressing
+			if strings.Contains(returnErr.Error(), "some StorageClasses were skipped while waiting for pre-requisites to be met") {
+				if instance.Status.Phase != statusutil.PhaseOnboarding {
+					instance.Status.Phase = statusutil.PhaseProgressing
+				}
+				return reconcile.Result{Requeue: true}, nil
+			}
+
 			if instance.Status.Phase != statusutil.PhaseOnboarding {
 				instance.Status.Phase = statusutil.PhaseError
 			}
```

Comment:
```suggestion
			if instance.Status.Phase != statusutil.PhaseOnboarding {
			        // if the error was due to skipped storage classes, we shouldn't set the phase to error
			        // instead of error phase we will set it to progressing
			        if strings.Contains(returnErr.Error(), "some StorageClasses were skipped while waiting for pre-requisites to be met") {         
			                instance.Status.Phase = statusutil.PhaseProgressing
			                return reconcile.Result{Requeue: true}, nil
			        }
				instance.Status.Phase = statusutil.PhaseError
			}
```
I think this would be a better way to write it.

Reply:
Yeah, I agree. Didn't think of this approach, but it looks cleaner. Thanks.
Have made the changes.